seismic/inventory/add_time_corrections.py

```diff
@@ -0,0 +1,154 @@
+#! /usr/bin/env python
+"""
+Add GPS clock time correction csv_data into inventory file to get a modified station xml file
+
+CreationDate:
+    24/02/2020
+
+Developer:
+    fei.zhang@ga.gov.au
+
+"""
+
+import os
+import sys
+from obspy import read_inventory
+from obspy import Inventory
+from obspy.core.inventory import Network
+from obspy.core.util import AttribDict
+
+
+def get_csv_correction_data(path_csvfile):
+    """
+    Read in the csv data from an input file, get the network_code, station_code, csv_data
+
+    Args:
+        path_csvfile: input csv file in /g/data/ha3/Passive/SHARED_DATA/GPS_Clock/corrections/
+        $ head 7D.DE43_clock_correction.csv
+        net,sta,date,clock_correction
+        7D,DE43,2012-11-27,1.0398489013215846
+        7D,DE43,2012-11-28,0.9408504322549281
+        7D,DE43,2012-11-29,0.8418519631882714
+        7D,DE43,2012-11-30,0.7428534941216148
+        7D,DE43,2012-12-01,0.6438550250549583
+
+    Returns: (network_code, station_code, csv_data)
+
+    """
+    with open(path_csvfile, "r") as csvfid:
+        all_lines = csvfid.readlines()
+
+    print("The csv file length", len(all_lines))
+    print("the first line", all_lines[0])
+    print("the last  line", all_lines[-1])
+
+    line2 = all_lines[1]
+    print("line2=", line2)
+
+    my_items = line2.split(",")
+
+    network_code = my_items[0]  # network_code = "7D"
+    station_code = my_items[1]  # station_code = "DE43"
+
+    return (network_code, station_code, all_lines)
+
+
+def add_gpscorrection_into_stationxml(csv_file, input_xml, out_xml=None):
+    """
+    read in the correction CSV data from a file, get the station metadata node from input_xml file, 
+    then add the CSV data into the station xml node to write into out_xml
+    Args:
+        csv_file: input csv file with correction data
+        input_xml: input original stationXML file which contains the metadata for the network and station of csv_file
+        out_xml:  Directory of the output xml file
```

Comment:
Auto-naming the output file in this function like this tends to create problems with scaling up to processing large batches of files. We can't make any assumptions about the granularity or content of the input xml file, so we shouldn't make assumptions about suitable file naming. Consider fully delegating the output file naming to the calling code, and change `out_xml` from a directory to be just the actual output file name.

Reply:
Please comment.

Reply:
the output xml file name use its station code, which can be conveniently obtained by program. Push the out xml file name to calling code means that you have to do  similar things. 
This need to be reconsidered at refactoring stage.